external/torch-mlir/lib/Dialect/Torch/IR/TorchTypes.cpp

```diff
@@ -264,18 +269,30 @@ ValueTensorType::getWithLeastStaticInformation(MLIRContext *context) {
 }
 
 ValueTensorType ValueTensorType::getFromShaped(ShapedType type) {
-  return ValueTensorType::get(type.getContext(),
-                              type.hasRank() ? type.getShape()
-                                             : Optional<ArrayRef<int64_t>>(),
-                              type.getElementType());
+  Type dtype = type.getElementType();
+  if (auto integerType = dtype.dyn_cast<IntegerType>()) {
+    if (integerType.isSignless())
+      dtype = IntegerType::get(type.getContext(), integerType.getWidth(),
+                               IntegerType::Signed);
+  }
+
+  return ValueTensorType::get(
+      type.getContext(),
+      type.hasRank() ? type.getShape() : Optional<ArrayRef<int64_t>>(), dtype);
 }
 
 TensorType ValueTensorType::toBuiltinTensor() const {
   if (!hasDtype())
     return nullptr;
   if (!hasSizes())
     return UnrankedTensorType::get(getDtype());
-  return RankedTensorType::get(getSizes(), getDtype());
+  Type dtype = getDtype();
+  if (auto integerType = dtype.dyn_cast<IntegerType>()) {
```

Comment:
nit: can you put this in a separate function `convertDtypeToBuiltinElementType`, and have it handle an explicit allowlist (probably int and float for now. qint8 and stuff we can add later)